packages/devextreme/js/__internal/ui/chat/chat_message_list.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import $ from '@js/core/renderer';
+import type { Message } from '@js/ui/chat';
+import type { WidgetOptions } from '@js/ui/widget/ui.widget';
+
+import Widget from '../widget';
+import MessageGroup from './chat_message_group';
+
+const CHAT_MESSAGE_LIST_CLASS = 'dx-chat-message-list';
+const CHAT_MESSAGE_LIST_CONTENT_CLASS = 'dx-chat-message-list-content';
+
+export interface MessageListOptions extends WidgetOptions<MessageList> {
+  items?: Message[];
+  currentUserId?: string;
+}
+
+class MessageList extends Widget<MessageListOptions> {
+  _getDefaultOptions(): MessageListOptions {
+    return {
+      ...super._getDefaultOptions(),
+      ...{
+        items: [],
+        currentUserId: '',
+      },
+    };
+  }
+
+  _initMarkup(): void {
+    $(this.element()).addClass(CHAT_MESSAGE_LIST_CLASS);
+
+    super._initMarkup();
+
+    this._renderMessageListContent();
+  }
+
+  _isCurrentUser(id): boolean {
+    const { currentUserId } = this.option();
+
+    return currentUserId === id;
+  }
+
+  _messageGroupAlignment(id): 'start' | 'end' {
+    return this._isCurrentUser(id) ? 'end' : 'start';
```

Comment:
I don't sure that in rtl we should automatically inverse alignment of messages in chat. So, seems in this case it shouldn't be 'end', 'start'.

Reply:
> I'm not sure that in rtl we should automatically inverse alignment of messages in chat
That was never in question. Where did you get that from?

Block start and block end is an appropriate, generalized name for it.